icekit/templatetags/icekit_tags.py

```diff
@@ -359,3 +361,27 @@ def sharedcontent_exists(slug):
     from fluent_contents.plugins.sharedcontent.models import SharedContent
     site = Site.objects.get_current()
     return SharedContent.objects.parent_site(site).filter(slug=slug).exists()
+
+
+@register.inclusion_tag(
+    'icekit/navigation/navigation.html',
+    name='render_navigation',
+    takes_context=True,
+)
+def render_navigation(context, identifier):
+    request = context['request']
+
+    navigation_slug = slugify(identifier)
+    navigation = navigation_models.Navigation.objects.filter(slug=navigation_slug).first()
+    if navigation:
+        active_navigation_items = navigation.get_active_items_for_request(request)
```

Comment:
I think it would be cleaner to make `active_items` a property of `navigation` rather than set a whole new context variable. The problem is passing `request` around.

I suggest instead calling `navigation.set_request(request)` here, and calling `active_items` from the template.